### DISCUSSION TITLE
How can I declare an argument without a value with MyST-Parser / Sphinx directive?

### DISCUSSION THREAD

**Discussion by zakhenry** [executablebooks/meta#492] [Q&A]
e.g. I want to use the [automatic dedent](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html?highlight=dedent#directive-option-code-block-dedent) feature of literalinclude:

````
```{literalInclude} path/to/my/file
---
linenos: true
lines: 19-36
lineno-match: true
dedent:
---
```

````
With the above I get the following error:

```
WARNING: Directive 'literalInclude': Invalid option value: (option: 'dedent'; value: )
invalid literal for int() with base 10: ''

---
linenos: true
lines: 19-36
lineno-match: true
dedent: 
---
```

and if I try just `dedent` I get 

```
WARNING: Directive 'literalInclude': Invalid options YAML: while scanning a simple key
  in "<unicode string>", line 4, column 1:
    dedent
    ^
could not find expected ':'
  in "<unicode string>", line 5, column 1:
    
    ^

---
linenos: true
lines: 19-36
lineno-match: true
dedent
---

```

I've also tried `dedent: None` and `dedent: ''` and `dedent: ""` with no luck

How do I declare an optional but empty argument?

I'm not sure but this might be a bug in MyST-Parser; in the Sphinx definition the test is using `None` to test the feature, see https://github.com/sphinx-doc/sphinx/pull/8684/files

however in MyST-Parser, it coerces a None value into a string:

https://github.com/executablebooks/MyST-Parser/blob/master/myst_parser/parse_directives.py#L149-L150

**Comment by choldgraf** [comment]
It seems like you have correctly identified the problem 👍 

- `dedent` will allow `None` as an argument
- however MyST Parser casts `None` to `""`
- If a string is passed to `dedent` it tries to cast it as a number
- That causes the error
- Thus, it is not possible to pass `None` to `dedent`

I'm not sure how to fix this with the pre-existing MyST code other than to manually specify a number there. :-/

**Reply by zakhenry** [reply]
Great thanks, I'll raise this as a bug on the main repo 👍

**Reply by zakhenry** [reply]
raised https://github.com/executablebooks/MyST-Parser/issues/431
